src/test/groovy/com/palantir/gradle/shadowjar/ShadowJarPluginIntegrationSpec.groovy

```diff
@@ -400,10 +399,11 @@ class ShadowJarPluginIntegrationSpec extends IntegrationSpec {
     }
 
     private String dependenciesInPom() {
-        def pomXml = new XmlParser().parse(
-                new File(projectDir, "${MAVEN_ROOT}/com/palantir/bar-baz_quux/asd-fgh/2/asd-fgh-2.pom"))
+        def pomFile = new File(projectDir, "${MAVEN_ROOT}/com/palantir/bar-baz_quux/asd-fgh/2/asd-fgh-2.pom")
+        def pomXml = new groovy.xml.XmlParser().parse(pomFile)
         // GCV publishes the shaded constraints in dependencyManagement - this should be fine
-        def dependenciesText = pomXml.dependencies.collect {XmlUtil.serialize(it)}.join('\n')
+        // Explicitly cast to "Node" before serializing to avoid errors due to Groovy choosing the wrong method override
```

Comment:
I was wondering when we'd hit the groovy 3.0 issues :p That reminds me, with Gradle 7/Groovy 3 we can start using `.stripIndent(true)` in our tests and have them work on a 15 JVM without weird formatting issues.